git-gui/git-gui.sh

```diff
@@ -2239,6 +2241,8 @@ proc do_git_gui {} {
 
 		if {$old_GIT_DIR ne {}} {
 			set env(GIT_DIR) $old_GIT_DIR
+		} else {
+			unset env(GIT_DIR)
```

Comment:
This one seems unnecessary: the code before this says:

```tcl
 	set exe [list [_which git]]
 	if {$exe eq {}} {
 		error_popup [mc "Couldn't find git gui in PATH"]
 	} else {
 		global env
 
 		if {[info exists env(GIT_DIR)]} {
 			set old_GIT_DIR $env(GIT_DIR)
 			unset env(GIT_DIR)
 		} else {
 			set old_GIT_DIR {}
 		}
 
 		set pwd [pwd]
 		cd $current_diff_path
 
 		eval exec $exe gui &
 ```

The relevant part is:

```tcl
		if {[info exists env(GIT_DIR)]} {
 			set old_GIT_DIR $env(GIT_DIR)
 			unset env(GIT_DIR)
 		} else {
 			set old_GIT_DIR {}
 		}
 ```

In other words, if there is no `GIT_DIR` in the environment, `old_GIT_DIR` will be set to the empty string, and in this case this PR tries to unset it. But nothing in-between sets `GIT_DIR`, therefore this is a no-op.

Do you agree? If yes, please amend this commit to drop this hunk and force-push the branch to update this Pull Request.